tests/ignite/distributed/comp_models/test_native.py

```diff
@@ -499,31 +509,56 @@ def _test__native_dist_model_spawn(backend, num_workers_per_machine, device, ini
-@pytest.mark.parametrize("init_method", [None, "env://", "tcp://0.0.0.0:22334", "FILE"])
+@pytest.mark.parametrize("init_method", [None, "CUSTOM_ADDR_PORT", "env://", "tcp://0.0.0.0:22334", "FILE"])
 def test__native_dist_model_spawn_gloo(init_method, dirname):
+    spawn_kwargs = {}
+
     if init_method == "FILE":
         init_method = f"file://{dirname}/shared"
+    elif init_method == "CUSTOM_ADDR_PORT":
+        init_method = None
+        spawn_kwargs["master_addr"] = "0.0.0.0"
+        spawn_kwargs["master_port"] = 2345
 
     nproc = torch.cuda.device_count() if torch.cuda.is_available() else 4
     device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
-    _test__native_dist_model_spawn("gloo", num_workers_per_machine=nproc, device=device, init_method=init_method)
+    _test__native_dist_model_spawn(
+        "gloo", num_workers_per_machine=nproc, device=device, init_method=init_method, **spawn_kwargs
+    )
     if device.type == "cpu":
+        spawn_kwargs["start_method"] = "fork"
         _test__native_dist_model_spawn(
-            "gloo", num_workers_per_machine=nproc, device=device, start_method="fork", init_method=init_method
+            "gloo", num_workers_per_machine=nproc, device=device, init_method=init_method, **spawn_kwargs
         )
 
+    if init_method not in [None, "env://"]:
+        with pytest.raises(ValueError, match=r"master_addr should be None if init_method is provided"):
+            _test__native_dist_model_spawn(
+                "gloo", num_workers_per_machine=nproc, device=device, init_method=init_method, master_addr="abc"
+            )
+        with pytest.raises(ValueError, match=r"master_port should be None if init_method is provided"):
+            _test__native_dist_model_spawn(
+                "gloo", num_workers_per_machine=nproc, device=device, init_method=init_method, master_port=123
+            )
+
 
 @pytest.mark.distributed
 @pytest.mark.skipif("WORLD_SIZE" in os.environ, reason="Skip if launched as multiproc")
 @pytest.mark.skipif(torch.cuda.device_count() < 1, reason="Skip if no GPU")
-@pytest.mark.parametrize("init_method", [None, "tcp://0.0.0.0:22334", "FILE"])
+@pytest.mark.parametrize("init_method", [None, "CUSTOM_ADDR_PORT", "tcp://0.0.0.0:22334", "FILE"])
 def test__native_dist_model_spawn_nccl(init_method, dirname):
+    spawn_kwargs = {}
+
     if init_method == "FILE":
         init_method = f"file://{dirname}/shared"
+    elif init_method == "CUSTOM_ADDR_PORT":
+        init_method = None
+        spawn_kwargs["master_addr"] = "0.0.0.0"
+        spawn_kwargs["master_port"] = 2345
 
-    num_workers_per_machine = torch.cuda.device_count()
+    nproc = torch.cuda.device_count()
     _test__native_dist_model_spawn(
-        "nccl", num_workers_per_machine=num_workers_per_machine, device="cuda", init_method=init_method
+        "nccl", num_workers_per_machine=nproc, device="cuda", init_method=init_method, **spawn_kwargs
     )
 
```

Comment:
may add also this like in the `gloo` case but it's a bit redundant ... ?
```python
if init_method not in [None, "env://"]:
    with pytest.raises(ValueError, match=r"master_addr should be None if init_method is provided"):
        _test__native_dist_model_spawn(
            "nccl",   ... , master_addr="abc"
        )
    with pytest.raises(ValueError, match=r"master_port should be None if init_method is provided"):
        _test__native_dist_model_spawn(
            "nccl", ... ,  master_port=123
        )
```

Reply:
yes, it is redundant and IMO does not depend on the backend, this do not check for all backends.